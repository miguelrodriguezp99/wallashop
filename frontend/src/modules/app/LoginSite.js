import React from "react";
import LoginForm from "./components/forms/LoginForm";

const LoginSite = () => {
	return (
		<>
			<LoginForm />
		</>
	);
};

export default LoginSite;
