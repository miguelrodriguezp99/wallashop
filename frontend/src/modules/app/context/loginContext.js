import { createContext, useState } from "react";

// Este es el que tenemos que consumir
export const LogginContext = createContext();

// Este es el que nos provee de acceso al contexto
export function LogginProvider({ children }) {
	const [token, setToken] = useState(null);
	const [image, setImage] = useState(null);
	const [user, setUser] = useState({});

	return (
		<LogginContext.Provider
			value={{
				token,
				setToken,
				image,
				setImage,
				user,
				setUser,
			}}
		>
			{children}
		</LogginContext.Provider>
	);
}
