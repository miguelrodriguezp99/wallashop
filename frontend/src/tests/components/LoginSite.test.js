import renderer from 'react-test-renderer';
import LoginSite from '../../modules/app/LoginSite';
import { LogginContext } from "../../modules/app/context/loginContext"; // AsegÃºrate de usar la ruta correcta
import { MemoryRouter } from "react-router-dom";

const fakeLogginContext = {
	token: "testToken",
};

describe('LoginSite component', () => {
    it('renders correctly', () => {
      const tree = renderer.create(
        <MemoryRouter>
            <LogginContext.Provider value={fakeLogginContext}>
                <LoginSite />
            </LogginContext.Provider>
        </MemoryRouter>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
});