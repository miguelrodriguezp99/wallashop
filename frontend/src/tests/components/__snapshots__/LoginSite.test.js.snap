// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginSite component renders correctly 1`] = `
<div
  className="flex items-center justify-center overflow-hidden h-[calc(100vh-65px)]"
>
  <div
    className="border border-solid border-gray-200 rounded-2xl text-center text-gray-200 font-helvetica py-10 px-16 w-300 mb-80 bg-black mt-60 overflow-hidden hover:border-blue-300"
  >
    <h1
      className="text-7xl font-bold bg-gradient-to-r from-blue-500 to-blue-700 text-transparent bg-clip-text"
    >
      LOG IN
    </h1>
    <form
      onSubmit={[Function]}
    >
      <label>
        <div
          className="form__group field"
        >
          <input
            className="form__field"
            name="username"
            onChange={[Function]}
            placeholder="Username"
            required=""
            type="text"
            value=""
          />
          <label
            className="form__label"
            htmlFor="name"
          >
            Username
          </label>
        </div>
      </label>
      <label>
        <div
          className="form__group field"
        >
          <input
            className="form__field"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            required=""
            type="password"
            value=""
          />
          <label
            className="form__label"
            htmlFor="name"
          >
            Password
          </label>
        </div>
      </label>
      <div
        className="form-buttons"
      >
        <button
          className="group tap-highlight-transparent transition-transform-colors-opacity data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-medium bg-transparent px-unit-4 min-w-unit-20 h-unit-10 text-small gap-unit-2 rounded-medium [&>svg]:max-w-[theme(spacing.unit-8)] data-[pressed=true]:scale-[0.97] transition-transform-colors-opacity motion-reduce:transition-none border-primary hover:!text-primary-foreground hover:!bg-primary text-white ml-2 mr-2"
          disabled={false}
          href="#"
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
          type="button"
        >
          Log in
        </button>
        <button
          className="group tap-highlight-transparent transition-transform-colors-opacity data-[hover=true]:opacity-hover z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-medium bg-transparent px-unit-4 min-w-unit-20 h-unit-10 text-small gap-unit-2 rounded-medium [&>svg]:max-w-[theme(spacing.unit-8)] data-[pressed=true]:scale-[0.97] transition-transform-colors-opacity motion-reduce:transition-none border-primary hover:!text-primary-foreground hover:!bg-primary text-white ml-2 mr-2"
          disabled={false}
          href="#"
          onBlur={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
          type="button"
        >
          Home
        </button>
      </div>
    </form>
  </div>
</div>
`;
